File: LC-217-Contains-Duplicate-Arrays-Hashing/LC-217-contains_duplicate-arrays_hashing.js
/*
Given an integer array 'nums', return True if any value 
appears at least twice in the array, and return False if
every element is distinct.
*/

var containsDuplicate = function (nums) {
  let counts = {};

  for (let i = 0; i < nums.length; i++) {
    num = nums[i];
    if (counts.hasOwnProperty(num)) {
      return true;
    } else {
      counts[num] = 1;
    }
  }
  return false;
};

File: LC-217-Contains-Duplicate-Arrays-Hashing/LC-217-contains_duplicate-arrays_hashing.py
'''
Given an integer array 'nums', return True if any value 
appears at least twice in the array, and return False if
every element is distinct.
'''

class Solution:
    def containsDuplicate(self, nums) -> bool:
        num_counts = {}
        for num in nums:
            if num not in num_counts:
                num_counts[num] = 1
            else:
                return True
        
        return False











